src/context/FilterProjectsContext.jsx

```diff
@@ -171,110 +176,133 @@ export const FilterProjectsProvider = ({ children }) => {
+
+      return projectData.results
+        .map((project) => {
+          // Filter project if sample date falls within selected date range
+          if (!sampleDateAfter && !sampleDateBefore) {
+            return project
+          }
+          const beginDate = sampleDateAfter || fallbackSampleDateAfter
+          const finishDate = sampleDateBefore || fallbackSampleDateBefore
+          return {
+            ...project,
+            records: project.records.filter((record) => {
+              const recordDate = new Date(record.sample_date)
+              return recordDate <= new Date(finishDate) && recordDate >= new Date(beginDate)
+            }),
+          }
+        })
+        .map((project) => {
+          // Filter projects based on data sharing policy
+          if (dataSharingFilter) {
+            const policies = [
+              'data_policy_beltfish',
+              'data_policy_benthiclit',
+              'data_policy_benthicpit',
+              'data_policy_benthicpqt',
+              'data_policy_bleachingqc',
+              'data_policy_habitatcomplexity',
+            ]
+            return {
+              ...project,
+              records: project.records.filter((record) =>
+                policies.some((policy) => record[policy] === 'public summary'),
+              ),
+            }
+          } else {
+            return project
+          }
+        })
+        .map((project) => {
+          // Filter projects based on collection method
+          if (methodFilters.length === 0) {
+            return project
+          } else {
+            return {
+              ...project,
+              records: project.records.filter((record) =>
+                methodFilters.some((method) => Object.keys(record.protocols).includes(method)),
+              ),
+            }
+          }
+        })
+        .filter((project) => {
+          // Filter by selected countries
+          const matchesSelectedCountries =
+            selectedCountries.length === 0 ||
+            selectedCountries.includes(project.records[0]?.country_name)
+
+          // Filter by selected organizations
+          const matchesSelectedOrganizations =
+            selectedOrganizations.length === 0 ||
+            project.records[0]?.tags?.some((tag) => selectedOrganizations.includes(tag.name))
+
+          // Filter by project name
+          const matchesProjectName =
+            projectNameFilter === '' ||
+            project.records[0]?.project_name.toLowerCase().includes(projectNameFilter.toLowerCase())
+
+          // Filter out projects with empty records
+          const nonEmptyRecords = project.records.length > 0
+
+          // Filter out projects that the user is not a member of
+          const onlyShowProjectsUserIsAMemberOf = showYourData
+            ? userIsMemberOfProject(project.project_id, mermaidUserData)
+            : true
+
+          return (
+            matchesSelectedCountries &&
```

Comment:
I think it would be clearer to declare the returned values as a const with a clearer name and boolean value?